lib/much-rails/destroy_service.rb

```diff
@@ -18,12 +18,49 @@ module MuchRails::DestroyService
 
     around_call do |receiver|
       receiver.call
-    rescue MuchRails::Records::ValidateDestroy::DestructionInvalid => ex
+    rescue *MuchRails::DestroyService::ValidationErrors.exception_classes => ex
       set_the_return_value_for_the_call_method(
-        MuchRails::Result.failure(
-          exception: ex,
-          validation_errors: ex&.destruction_errors.to_h,
-        ),
+        MuchRails::DestroyService::ValidationErrors.result_for(ex),
+      )
+    end
+  end
+
+  module ValidationErrors
+    def self.add(exception_class, &block)
+      service_validation_errors.add(exception_class, &block)
+    end
+
+    def self.exception_classes
+      service_validation_errors.exception_classes
+    end
+
+    def self.result_for(ex)
+      service_validation_errors.result_for(ex)
+    end
+
+    def self.service_validation_errors
+      @service_validation_errors ||=
+        MuchRails::ServiceValidationErrors
+          .new
+          .tap do |e|
+            e.add(MuchRails::Records::DestructionInvalid) do |ex|
+              MuchRails::DestroyService::FailureResult.new(
```

Comment:
@kellyredding - I went with adding this helper instead of the whole `instance_eval` the block to get a helper method. I think this is simpler and, to me, provides the same function of providing a helper for building the result correctly.

Reply:
Nice!